src/services/external/excalidraw-backend/excalidraw.amqp.server.factory.provider.ts

```diff
@@ -0,0 +1,374 @@
+      transports: ['websocket', 'polling'],
+      allowEIO3: true,
+    });
+    // subscribe to the state of other server instances
+    // events from other servers ONLY
+    excalidrawEventSubscriber.subscribeToAll(event =>
+      event.handleEvent(wsServer)
+    );
+
+    wsServer.on(CONNECTION, async socket => {
+      const agentInfo = await getUserInfo(
+        kratosClient,
+        socket.handshake.headers,
+        logger,
+        authService
+      );
+
+      if (!agentInfo) {
+        closeConnection(socket, 'Not able to authenticate user');
+        return;
+      }
+
+      logger?.verbose?.(
+        `User '${agentInfo.userID}' established connection`,
+        LogContext.EXCALIDRAW_SERVER
+      );
+
+      wsServer.to(socket.id).emit(INIT_ROOM);
+      // client events ONLY
+      socket.on(
+        JOIN_ROOM,
+        async (roomID: string) =>
+          await joinRoomEventHandler(
+            roomID,
+            agentInfo,
+            socket,
+            wsServer,
+            whiteboardRtService,
+            authorizationService,
+            excalidrawEventPublisher,
+            logger
+          )
+      );
+
+      socket.on(SERVER_BROADCAST, (roomID: string, data: ArrayBuffer) =>
+        serverBroadcastEventHandler(
+          roomID,
+          data,
+          socket,
+          excalidrawEventPublisher
+        )
+      );
+
+      socket.on(
+        SERVER_VOLATILE_BROADCAST,
+        (roomID: string, data: ArrayBuffer) =>
+          serverVolatileBroadcastEventHandler(
+            roomID,
+            data,
+            socket,
+            excalidrawEventPublisher
+          )
+      );
+
+      socket.on(
+        DISCONNECTING,
+        async () =>
+          await disconnectingEventHandler(
+            agentInfo,
+            wsServer,
+            socket,
+            logger,
+            excalidrawEventPublisher
+          )
+      );
+
+      socket.on(DISCONNECT, () =>
+        disconnectEventHandler(socket, excalidrawEventPublisher)
+      );
+    });
+  } catch (error) {
+    logger.error(error, LogContext.EXCALIDRAW_SERVER);
+  }
+};
+
+/* Sets the user into the context field or closes the connection */
+const authenticate = async (
+  kratosFrontEndApi: FrontendApi,
+  headers: Record<string, string | string[] | undefined>,
+  logger: LoggerService,
+  authService: AuthenticationService
+): Promise<AgentInfo> => {
+  const cookie = headers.cookie as string;
+
+  try {
+    const { data: session } = await kratosFrontEndApi.toSession({
+      cookie,
+    });
+
+    if (!session) {
+      logger.verbose?.('No Ory Kratos session', LogContext.EXCALIDRAW_SERVER);
+      return authService.createAgentInfo();
+    }
+
+    const oryIdentity = session.identity as OryDefaultIdentitySchema;
+    return authService.createAgentInfo(oryIdentity);
+  } catch (e: any) {
+    throw new Error(e?.message);
+  }
+};
+/* returns the user agent info */
+const getUserInfo = async (
```

Comment:
`getUserInfo` looks like an util function, it may be good to split it into a util file and have other utilities there as well.